pkg/iptableswrapper/mocks/iptables_maps.go

```diff
@@ -91,6 +97,11 @@ func (ipt *MockIptables) List(table, chain string) ([]string, error) {
 	var chains []string
 	chainContents := ipt.DataplaneState[table][chain]
 	for _, ruleSpec := range chainContents {
+		// If a rule spec is just the chain name, it is the chain creation rule
+		if len(ruleSpec) == 1 {
+			chains = append(chains, fmt.Sprintf("-N %s", chain))
```

Comment:
The `List` API should not be modifying the chain. Shouldn't we be adding the `-N` entry when the chain is created? This is the part that is confusing me